### ISSUE TITLE
Initial state vs Rehydration

### ISSUE THREAD

**Issue by dlee-zailab** [zewa666/aurelia-store#28]
I'm a bit confused about the fact that initial state and localStorage rehydration seem at odds with one another. The plugin insists on an initial state and then promptly overwrites the localStorage with the initial state before any rehydration can take place. The workaround I've used is to manually read the localStorage state and set it as the initial state in main.ts. Is this how it's meant to work, or am I missing something obvious?

**Comment by zewa666** [comment]
Yep thats pretty much the out-of-the-box behaviour. You have to ways to work around that for now. Either, as you suggested, set the initialState directly from the localStorage, or register the localStorage middleware at a later time, e.g after the first emitted stream.

Both feel a bit inconvenient, that's true. What would you say if the localStorrageMiddleware would accept an additional setting besides `key`, something like `skipInitial` to actually do it's magic only if a previous state exists, that way effectively skipping the first turn?

**Comment by dlee-zailab** [comment]
Thanks so much for the prompt reply! I think your suggestion of registering the middleware after the initialisation is a good one, I will give it a try. Thanks for a great plugin, by the way :)

**Comment by dlee-zailab** [comment]
So your suggestion worked really well. It was very useful to be able to determine where in my app lifecycle I wanted the localStorage middleware to kick in so none of my rehydration was affected. I'm not sure if adding `skipInitial` is as useful as just being aware that you can add middleware "on the fly". That's very powerful.

**Comment by mabouchalkha** [comment]
Can you show an example please thank you

**Comment by zewa666** [comment]
@mabouchalkha well essentially you either put the `registerMiddlware` call to a later moment, wherever that might be in your sample. Or via a new subscription:

```typescript
store.state.pipe(skip(1)).subscribe(() => {
  this.store.registerMiddleware(...);
});
```

**Comment by mabouchalkha** [comment]
@zewa666 thank you very powerful plugin

**Comment by zewa666** [comment]
@dlee-zailab any further comments on this one or can we close this?
